cirq/ops/common_gates.py

```diff
@@ -187,7 +188,10 @@ def known_qasm_output(self,
     def __str__(self) -> str:
         if self.half_turns == 1:
             return 'X'
-        return 'X**{!r}'.format(self.half_turns)
+        elif isinstance(self.half_turns, float):
+            return 'X**{!r}'.format(self.half_turns)
+        return 'X**symbol.Symbol({!r})'.format(
```

Comment:
Does the `repr` of a sympy expression not return evaluable code? I would expect `X**{!r}` to be sufficient.

Reply:
Could you please elaborate on your question?

Reply:
If `repr(sympy_thing)` returns code that evaluates to something equal to `sympy_thing`'s value, then `"cirq.X**{!r}`.format(sympy_thing)` should be code that evaluates to something equal to the gate. There shouldn't be any `symbol.Symbol` because, well, that class is not going to exist anymore so that's not evaluable code.